core/Tracker/GoalManager.php

```diff
@@ -195,6 +207,42 @@ public function detectGoalMatch($goal, Action $action)
         return $action->getActionUrl();
     }
 
+    private function detectNumericGoalMatch($goal, Action $action, VisitProperties $visitProperties, Request $request)
+    {
+        switch ($goal['match_attribute']) {
+            case 'visit_duration':
+                $firstActionTime = $visitProperties->getProperty('visit_first_action_time');
+                if (empty($firstActionTime)) {
+                    return null;
+                }
+
+                $visitDurationInSecs = $request->getCurrentTimestamp() - ((int) $firstActionTime);
```

Comment:
I haven't tested but wouldn't `visit_first_action_time ` be a date like `2019-02-03 04:05:06`?